Sources/Persistence/CoreData/NSManagedObjectContext+CoreDataStack.swift

```diff
@@ -36,54 +47,71 @@ public extension NSManagedObjectContext {
 public extension NSManagedObjectContext {
 
     func persistChanges(_ description: String, saveParent: Bool = true, waitForParent: Bool = false) throws {
+
         guard hasChanges else { return }
 
         do {
             try save()
         } catch let error {
             // FIXME: use proper logging function when available
-            print("Error saving context \(self) (\(description)): \(error). Rollbacking all changes...")
+            print("💥[Alicerce.NSManagedObjectContext]: Error saving context \(self) (\(description)): \(error). " +
+                  "Rolling back all changes...")
 
             rollback()
             throw error
         }
 
         guard saveParent, let parent = parent else { return }
 
+        var parentError: Error?
+
         let parentSave = {
             guard parent.hasChanges else { return }
 
             do {
                 try parent.save()
             } catch let error {
                 // FIXME: use proper logging function when available
-                print("Error saving parent context \(parent) (\(description)): \(error). Rollbacking all changes...")
+                print("💥[Alicerce.NSManagedObjectContext]: Error saving parent context \(self) (\(description)): " +
+                      "\(error). Rolling back all changes...")
                 parent.rollback()
+                parentError = error
             }
         }
 
         waitForParent
             ? parent.performAndWait(parentSave)
             : parent.perform(parentSave)
+
+        if let parentError = parentError {
+            throw parentError
+        }
     }
 }
 
 // MARK: - Parent Coordinator & Store type
 
 public extension NSManagedObjectContext {
 
-    func topLevelPersistentStoreCoordinator() -> NSPersistentStoreCoordinator {
+    var topLevelPersistentStoreCoordinator: NSPersistentStoreCoordinator? {
+
         switch (persistentStoreCoordinator, parent) {
-        case let (persistentStoreCoordinator?, _): return persistentStoreCoordinator
-        case let (_, parent?): return parent.topLevelPersistentStoreCoordinator()
-        default: fatalError("💥: Context doesn't have neither a `persistentStoreCoordinator` nor a `parent`!")
+        case (let persistentStoreCoordinator?, _):
+            return persistentStoreCoordinator
+        case (_, let parent?):
+            return parent.topLevelPersistentStoreCoordinator
+        default:
+            return nil
         }
     }
 
-    func isSQLiteStoreBased() -> Bool {
-        switch topLevelPersistentStoreCoordinator().firstStoreType() {
-        case .sqLite: return true
-        default: return false
+    var isSQLiteStoreBased: Bool {
+
+        switch topLevelPersistentStoreCoordinator?.firstStoreType {
+        case .sqLite?:
```

Comment:
As discussed, maybe rethink capitalization in `.sqLite`

Reply:
sqlite?

Reply:
`sqlite`?

Reply:
either that or `SQLite`, any preference? 🚲